cohttp-lwt/src/server.ml

```diff
@@ -138,10 +138,7 @@ module Make (IO : S.IO) = struct
     let handler request =
       match request with
       | None ->
-          let rec sleep_till_cancelled () =
-            Lwt_unix.sleep 1. >>= fun () -> sleep_till_cancelled ()
-          in
-          sleep_till_cancelled ()
+          fst @@ Lwt.wait ()
```

Comment:
Lwt.wait() is uncancellable. You need to use Lwt.task() which is cancelleable.
[Lwt doc](https://ocsigen.org/lwt/latest/manual/manual) says:
"The last one, `pick`, is the same as `choose`, except that it tries to cancel all other promises when one resolves. Promises created via `Lwt.wait()` are not cancellable and are thus not cancelled."

Reply:
I see, it is fun though that the server still closes when I use `Lwt.wait ()`..